collect/collect.go

```diff
@@ -289,6 +289,21 @@ func Sample(metric string, ts opentsdb.TagSet, v float64) error {
 	return nil
 }
 
+// StartTimer records the current time, and returns a function you can call to
+// record the end of your action.
+//
+// Typical usage would be:
+//    done := collect.StartTimer("myMetric", opentsdb.TagSet{})
+//    doMyThing()
+//    done()
+func StartTimer(metric string, ts opentsdb.TagSet) func() {
```

Comment:
Benchmark tests are great for build-time benchmarks. It can't be used for run-time instrumentation like this is doing.